src/plugins/intel_npu/src/plugin/npuw/llm_eagle3_extension.cpp

```diff
@@ -0,0 +1,197 @@
+        return;
+    }
+
+    // It's an Eagle3 model, now determine if it's Draft or Target based on inputs/outputs
+    bool has_hidden_states_input = in_ports.find(Eagle3LayerNames::hidden_states) != in_ports.end();
+    bool has_internal_hidden_states_input = in_ports.find(Eagle3LayerNames::internal_hidden_states) != in_ports.end();
+    bool has_last_hidden_state_output = out_ports.find(Eagle3LayerNames::last_hidden_state) != out_ports.end();
+
+    if (has_hidden_states_input && has_internal_hidden_states_input && has_last_hidden_state_output) {
+        m_role = Eagle3ModelRole::Draft;
+        LOG_INFO("Eagle3 Draft Model detected");
+    } else if (!has_hidden_states_input && !has_internal_hidden_states_input && has_last_hidden_state_output) {
+        m_role = Eagle3ModelRole::Target;
+        LOG_INFO("Eagle3 Target Model detected");
+    } else {
+        m_role = Eagle3ModelRole::None;
+        LOG_WARN("Eagle3 model flag set in rt_info, but model structure doesn't match Draft or Target pattern");
+    }
+}
+
+void Eagle3Extension::prepare_inputs(std::shared_ptr<ov::IAsyncInferRequest> request,
+                                     const std::unordered_map<std::string, ov::Output<const ov::Node>>& in_ports) {
+    // Only draft models need to prepare Eagle3 inputs
+    if (m_role != Eagle3ModelRole::Draft) {
+        return;
+    }
+
+    auto hidden_states_it = in_ports.find(Eagle3LayerNames::hidden_states);
+    if (hidden_states_it != in_ports.end() && m_hidden_states) {
+        auto padded_hidden_states = request->get_tensor(hidden_states_it->second);
+        pad_hidden_state_input(padded_hidden_states, m_hidden_states);
+        LOG_VERB("Eagle3 Draft: Set hidden_states input tensor");
+    }
+
+    auto internal_hidden_states_it = in_ports.find(Eagle3LayerNames::internal_hidden_states);
+    if (internal_hidden_states_it != in_ports.end() && m_internal_hidden_states) {
+        auto padded_internal_hidden_states = request->get_tensor(internal_hidden_states_it->second);
+        pad_hidden_state_input(padded_internal_hidden_states, m_internal_hidden_states);
+        LOG_VERB("Eagle3 Draft: Set internal_hidden_states input tensor");
+    }
+}
+
+void Eagle3Extension::process_outputs(std::shared_ptr<ov::IAsyncInferRequest> request,
+                                      const std::unordered_map<std::string, ov::Output<const ov::Node>>& out_ports) {
+    // Both draft and target models have last_hidden_state output
+    if (m_role == Eagle3ModelRole::Draft || m_role == Eagle3ModelRole::Target) {
+        auto last_hidden_state_it = out_ports.find(Eagle3LayerNames::last_hidden_state);
+        if (last_hidden_state_it != out_ports.end()) {
+            m_last_hidden_state = request->get_tensor(last_hidden_state_it->second);
+            LOG_VERB("Eagle3 " << (m_role == Eagle3ModelRole::Draft ? "Draft" : "Target")
+                               << ": Retrieved last_hidden_state output tensor");
+        }
+    }
```

Comment:
This function doesn't seem to do any type of processing for output, it just retrieves and stores it. Might be it can be named something like `update_last_hidden_state()` or something like this.